### ISSUE TITLE
Folder share name created by vagrant up is not deleted after vagrant destroy

### ISSUE THREAD

**Issue by unklhe** [hashicorp/vagrant#8418]
When I use vagrant up to create a VM in my vCenter, vagrant will enable the local folder containing the vagrantfile to be shared and give it a share name to be mounted to VM via SMB. When I use vagrant destroy, the share name should be removed,  as should be the default teardown operation. However the share name is not deleted. So after times of up/destroy, there are many share name left.

![image](https://cloud.githubusercontent.com/assets/5634509/24392616/4d6869e0-13c7-11e7-89fb-4deb5804a3f7.png)

### Vagrant version
Vagrant 1.9.2

### Host operating system
Windows 10

### Guest operating system
Centos7

### Vagrantfile
```ruby
ENV['VAGRANT_DEFAULT_PROVIDER'] = 'vcenter'

nodes = []

[*1..1].each do |n|
  nodes << { hostname: "centos#{n}",
             box: 'centos/7',
             ip: "10.200.200.20#{n}",
             mem: 1024 * n,
             cpu: n }
end


Vagrant.configure('2') do |config|

  # Go through nodes and configure each of them.
  nodes.each do |node|

    config.vm.provider :vcenter do |vcenter|
      vcenter.hostname = '192.168.3.10'
      vcenter.username = ENV['VCENTERUSER']
      vcenter.password = ENV['VCENTERPASSWORD']
      vcenter.folder_name = 'Test'
      vcenter.datacenter_name = 'DC1'
      vcenter.compute_name = 'Host1'
      vcenter.datastore_name = 'Datastore_1'
      vcenter.template_folder_name = 'Templates'
      vcenter.network_name = 'VM Network'
      vcenter.linked_clones = true
    end

    

    config.vm.define node[:hostname] do |node_config|
      node_config.vm.box = node[:box]
      node_config.vm.hostname = node[:hostname]
 
      # Let's configure the network for the VM, only the ip changes and is
      # coming from the nodes array
      node_config.vm.network :public_network,
                             ip: node[:ip],
                             netmask: '255.255.255.0',
                             gateway: '10.200.200.1',
                             dns_server_list: ['192.168.4.10'],
                             dns_suffix_list: ['test.local']

      # Let's override some provider settings for specific VMs
      node_config.vm.provider :vcenter do |override|
        # Override number of cpu and memory based on what's in the nodes array
        override.num_cpu = node[:cpu]
        override.memory = node[:mem]
        override.folder_name = 'Test'        
      end
      node_config.nfs.functional = false
    end
  end
end
```

### Debug output
https://gist.github.com/unklhe/c61a5b91653ff0f5b1519a9d249710a4

### Expected behavior
Folder share name should be removed after vagrant destroy

### Actual behavior
Folder share name is not removed

### Steps to reproduce
1. Open File Explorer and open //[host domain name]
2. vagrant up
3. back to File Explorer opened in step 1, a GUID formated share name can be found
4. vagrant destroy
5. back to File explorer in step 3. The share name is still there

### References
None

**Comment by chrisroberts** [comment]
This has been fixed as of Vagrant 2.0.2

Cheers!

**Comment by unklhe** [comment]
Glad to know. Cheers!
